### ISSUE TITLE
`nix-build -A exe -o result-exe` failing for existing and new project

### ISSUE THREAD

**Issue by cah6** [obsidiansystems/obelisk#197]
I have an existing project and, while `ob repl` and `ob run` succeed fine, `nix-build -A exe -o result-exe` is failing with the following:
```
error: attribute 'lens-family-th_0_4_1_0' missing, at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/configuration-ghc-7.10.x.nix:161:20
(use '--show-trace' to show detailed location information
```
while looking this up I found https://github.com/reflex-frp/reflex-platform/issues/273; does this mean that I can't use TH for lenses in my `./common` and `./frontend` subprojects? In any case I removed all usages of lens in both of those and ran this again but it resulted in the same error. To attempt to fix this I tried to add this version manually in my `default.nix`:
```
project ./. ({ pkgs, ... }: {
...
  overrides = self: super: {
    lens-family-th = self.callHackage "lens-family-th" "0.4.1.0" {};
  };
})
```
but I still don't fully understand nix expressions I'm writing and it resulted in:
```
$ nix-build -A exe -o result-exe
error: null pointer cast to ref
error: unexpected EOF reading a line
(use '--show-trace' to show detailed location information)
```
So then I tried making a completely new project and seeing if this works there. New project setup:
```
mkdir ob-test; cd ob-test
ob init; ob run;
```
and build still failed:
```
nix-build -A exe -o result-exe
these derivations will be built:
  /nix/store/g8k1sadrfhr71229nzv081yy1vf2v2nf-backend-0.1.drv
  /nix/store/h65iy36x07sj87mwsrp6xw5wywgdhhyy-frontend-0.1.drv
  /nix/store/gnmb1xrmqmr33sspdcjp06m5n9bsbkq3-compressedJs.drv
  /nix/store/jj9dwp6jlc5s7dwbjixn94vnc1vwa2dg-serverExe.drv
error: null pointer cast to ref
error: unexpected EOF reading a line
```
Edit: I'm on NixOS if that makes any difference.

**Comment by cah6** [comment]
Well, actually it looks like I succeeded in breaking everything in my setup. I tried running `nix-collect-garbage` in hopes that would fix this, but it didn't. Now even `ob init` and `ob run` are failing. For example:
```
$ nix-shell -A shells.ghc --show-trace
error: null pointer cast to ref
error: while evaluating anonymous function at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:144:30, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/attrsets.nix:199:52:
while evaluating 'workOnMulti'' at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:652:18, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:145:7:
while evaluating 'runCommand'' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/build-support/trivial-builders.nix:5:36, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:667:8:
while evaluating 'ghcWithPackages' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:182:23, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:665:10:
while evaluating 'withPackages' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:103:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:182:35:
while evaluating 'callPackage' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:22, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:103:28:
while evaluating 'callPackageWithScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:74:37, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:28:
while evaluating 'makeOverridable' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:72:24, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:97:8:
while evaluating 'drvScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:87:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:74:12:
while evaluating anonymous function at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/with-packages-wrapper.nix:1:1, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:87:27:
while evaluating 'closePropagation' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:228:22, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/with-packages-wrapper.nix:49:57:
while evaluating 'uniqList' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:159:14, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:228:29:
while evaluating 'go' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:160:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:166:8:
while evaluating 'innerClosePropagation' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:211:32, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:228:52:
while evaluating anonymous function at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:665:32, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:182:49:
while evaluating 'overiddenOut' at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:660:35, called from undefined position:
while evaluating 'overrideCabal' at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:186:26, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:660:39:
while evaluating the attribute 'backend' at /nix/store/8vqhlvifbm95y386qwjmxi0kj8bbp887-source/default.nix:276:21:
while evaluating 'addBuildDepend' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:41:25, called from /nix/store/8vqhlvifbm95y386qwjmxi0kj8bbp887-source/default.nix:276:38:
while evaluating 'addBuildDepends' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:42:26, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:41:28:
while evaluating 'overrideCabal' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:8:24, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:42:30:
while evaluating the attribute 'backend' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/attrsets.nix:199:44:
while evaluating anonymous function at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:141:35, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/attrsets.nix:199:52:
while evaluating 'callPackage' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:22, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:141:41:
while evaluating 'callPackageWithScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:74:37, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:28:
while evaluating 'makeOverridable' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:72:24, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:97:8:
while evaluating 'drvScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:87:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:74:12:
unexpected EOF reading a line$ nix-shell -A shells.ghc --show-trace
error: null pointer cast to ref
error: while evaluating anonymous function at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:144:30, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/attrsets.nix:199:52:
while evaluating 'workOnMulti'' at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:652:18, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:145:7:
while evaluating 'runCommand'' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/build-support/trivial-builders.nix:5:36, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:667:8:
while evaluating 'ghcWithPackages' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:182:23, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:665:10:
while evaluating 'withPackages' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:103:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:182:35:
while evaluating 'callPackage' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:22, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:103:28:
while evaluating 'callPackageWithScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:74:37, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:28:
while evaluating 'makeOverridable' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:72:24, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:97:8:
while evaluating 'drvScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:87:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:74:12:
while evaluating anonymous function at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/with-packages-wrapper.nix:1:1, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:87:27:
while evaluating 'closePropagation' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:228:22, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/with-packages-wrapper.nix:49:57:
while evaluating 'uniqList' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:159:14, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:228:29:
while evaluating 'go' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:160:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:166:8:
while evaluating 'innerClosePropagation' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:211:32, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/deprecated.nix:228:52:
while evaluating anonymous function at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:665:32, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:182:49:
while evaluating 'overiddenOut' at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:660:35, called from undefined position:
while evaluating 'overrideCabal' at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:186:26, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/default.nix:660:39:
while evaluating the attribute 'backend' at /nix/store/8vqhlvifbm95y386qwjmxi0kj8bbp887-source/default.nix:276:21:
while evaluating 'addBuildDepend' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:41:25, called from /nix/store/8vqhlvifbm95y386qwjmxi0kj8bbp887-source/default.nix:276:38:
while evaluating 'addBuildDepends' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:42:26, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:41:28:
while evaluating 'overrideCabal' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:8:24, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/lib.nix:42:30:
while evaluating the attribute 'backend' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/attrsets.nix:199:44:
while evaluating anonymous function at /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:141:35, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/attrsets.nix:199:52:
while evaluating 'callPackage' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:22, called from /nix/store/9wwm77gfgy7gznk1zm09nyhp5qm24z8d-source/project/default.nix:141:41:
while evaluating 'callPackageWithScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:74:37, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:101:28:
while evaluating 'makeOverridable' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:72:24, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:97:8:
while evaluating 'drvScope' at /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/pkgs/development/haskell-modules/make-package-set.nix:87:18, called from /nix/store/3ym8gkqkdg13gr12pa4s59k9j06wfd52-source/lib/customisation.nix:74:12:
unexpected EOF reading a line
```
I really have no idea what to do now as it seems like the most basic commands are failing. Any ideas would be appreciated!

**Comment by alexfmpe** [comment]
@cah6 mind having another go? @luigy seems to have fixed this

**Comment by ali-abrar** [comment]
Hi - please open a new issue if this persists. It looks like it was fixed via #200
